### DISCUSSION TITLE
Change default boot type to UEFI

### DISCUSSION THREAD

**Discussion by rp-** [apache/cloudstack#10136] [Q&A] [answered]
Is there a way to change the default boot type to UEFI/LEGACY?

The root of the problem I'm facing is, that the default `"pc"` machine type in rhel9 (`pc-i440fx-rhel7.6.0`) still doesn't support discard. So the only way to force the `q35` machine type (which supports discard) is by enabling `UEFI` boot type, but there seems to be no way to do this by default?

**Comment by weizhouapache** [comment]
AFAIK, it is only availble when deploy a vm
![image](https://github.com/user-attachments/assets/1784ce0a-b0cf-4014-b4da-ec9bee4e9a00)

currently boottype and boot mode cannot be set per template or per zone or globally.

**Comment by GutoVeronezi** [comment, marked as answer]
@rp-, after creating the VM, you can define the boot type via the VM settings. There is a setting called `UEFI` for which you can define the values `SECURE` or `LEGACY`.

![image](https://github.com/user-attachments/assets/6d003423-78b3-4bc0-a217-76ea401e5c44)

You can also define this setting on the template, making new VMs created with the template to inherit it.

The absence of the setting indicates that the VM will use `BIOS` + `LEGACY`.

**Reply by rp-** [reply]
Ah cool, I didn't know it would be inherited(or that you could set it) from the template!

**Reply by weizhouapache** [reply]
> @rp-, after creating the VM, you can define the boot type via the VM settings. There is a setting called `UEFI` for which you can define the values `SECURE` or `LEGACY`.
> 
> 
> You can also define this setting on the template, making new VMs created with the template to inherit it.
> 
> The absence of the setting indicates that the VM will use `BIOS` + `LEGACY`.

if template settings work, that's great
